### ISSUE TITLE
bug: Stencil should not require puppeteer dependency in new projects

### ISSUE THREAD

**Issue by sean-perkins** [ionic-team/stencil#4526]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/stencil/blob/main/CONTRIBUTING.md).
- [X] I agree to follow the [Code of Conduct](https://github.com/ionic-team/stencil/blob/main/CODE_OF_CONDUCT.md).
- [X] I have searched for [existing issues](https://github.com/ionic-team/stencil/issues) that already report this problem, without success.

### Stencil Version

v4

### Current Behavior

When creating a new project and removing the `puppeteer` dev dependency, Stencil will fail to start with the following exception:

```ts
[ ERROR ]  TypeScript: node_modules/@stencil/core/testing/jest/jest-environment.d.ts:9:44
           Cannot find module 'puppeteer' or its corresponding type declarations.

      L8:  setup(): Promise<void>;
      L9:  newPuppeteerPage(): Promise<import("puppeteer").Page>;
     L10:  closeOpenPages(): Promise<void>;

[ ERROR ]  TypeScript: node_modules/@stencil/core/testing/puppeteer/puppeteer-declarations.d.ts:3:120
           Cannot find module 'puppeteer' or its corresponding type declarations.

      L2:  import type { EventInitDict, EventSpy, ScreenshotDiff, ScreenshotOptions } from '@stencil/core/internal';
      L3:  ions, HTTPResponse, Page, ScreenshotOptions as PuppeteerScreenshotOptions, WaitForOptions } from 'puppeteer';
      L4:  /**

[27:32.9]  build failed, watching for changes... in 4.68 s
```

### Expected Behavior

Stencil should not require puppeteer as a dependency. Stencil projects may choose to use Playwright to test their component libraries and currently will require the dev dependency to satisfy Stencil. 

This worked previously in v2. Unsure if it regressed in v3 or v4. 

### System Info

```shell
System: node 16.15.1
    Platform: darwin (22.2.0)
   CPU Model: Apple M1 (8 cpus)
    Compiler: /Users/sean/Documents/ionic/issues/stencil-without-puppeteer/node_modules/@stencil/core/compiler/stencil.js
       Build: 1687953951
     Stencil: 4.0.1 ⛹
  TypeScript: 5.0.4
      Rollup: 2.42.3
      Parse5: 7.1.2
      Sizzle: 2.42.3
      Terser: 5.18.1
```

### Steps to Reproduce

1. Create a new project: `npm init stencil`
2. Remove the `puppeteer` dev dependency
3. Install dependencies
4. Run `npm run start`
5. Observe: Exception is thrown

### Code Reproduction URL

https://github.com/sean-perkins/stencil-puppeteer-bug

**Comment by rwaskiewicz** [comment]
It looks like this was introduced in v3.3.0, with the internal upgrade from Puppeteer v19->v20

**Comment by rwaskiewicz** [comment]
Upon further investigation, I believe this isn't a result of Puppeteer v20, but rather something related to TS 5.0 support (also added in v3.3.0)

**Comment by ayangweb** [comment]
Hi, did you resolve this issue? I'm currently creating a new project to install the dependencies, and it's also reporting this error!

**Comment by christian-bromann** [comment]
Hey @sean-perkins , a fix for this has been released in [`v4.8.2`](https://github.com/ionic-team/stencil/releases/tag/v4.8.2). Thanks for reporting!
